mono/metadata/mempool.c

```diff
@@ -137,7 +147,7 @@ mono_mempool_destroy (MonoMemPool *pool)
 void
 mono_mempool_invalidate (MonoMemPool *pool)
 {
-#ifdef MALLOC_ALLOCATION
+#ifdef INDIVIDUAL_ALLOCATIONS
```

Comment:
Checking-- when I refactored I tried not to change any actual behavior, so I kept in these places where we verify MALLOC_ALLOCATION is not being used in conjunction with these other features. Now that the two paths are unified however I think all of these features should actually work even in INDIVIDUAL_ALLOCATION mode. Could I go ahead and remove the g_assert_not_reached cases like this?